### PR TITLE
Add configurable mount point for secret engines

### PR DESCRIPTION
**Pull request by ckaipf**
## Objective

This pull request aims to enhance the functionality by introducing a new configuration parameter, `vault_secrets_mount_point`. The addition of this parameter will allow users to specify a different mount point for storing secrets, providing greater flexibility and organization within Vault.

## Updates

- Added the vault_secrets_mount parameter to the configuration and pass it to hvac.

### CODE REVIEW

File: src/fis/adapters/outbound/vault/client.py [hunk 1]
```diff
@@ -57,6 +57,11 @@ class VaultConfig(BaseSettings):
         description="Path without leading or trailing slashes where secrets should"
         + " be stored in the vault.",
     )
+    vault_secrets_mount_point: str = Field(
+        default="secret",
+        examples=["secret"],
```
Comment (mephenor) [inline src/fis/adapters/outbound/vault/client.py:62]:
```suggestion
        default="file-secrets",
        examples=["file-secrets"],
```
`secret` is a bit too generic, as everything we store is a secret.
Not sure if `file_secrets` might be better or not

**Review by mephenor — CHANGES_REQUESTED** [review]
One small thing regarding the config option.

Also, this should include at least a minor version bump, but probably even a major one, as changing the mount point is probably not backwards compatible

File: src/fis/adapters/outbound/vault/client.py (hunk 1 above)
Comment (ckaipf) [reply src/fis/adapters/outbound/vault/client.py:62]: I think the same. However, the default mount point for the HashiCorp Vault KV2 engine is `secret`. So I think we should leave it that way.

Comment (mephenor) [reply src/fis/adapters/outbound/vault/client.py:62]: So the default case would be backwards compatible with what we have right now?

**Review by mephenor — APPROVED** [review]

**Pull request merged**